docs/configuration.md

```diff
@@ -559,6 +559,39 @@ Apart from these, the following properties are also available, and may be useful
     <code>spark.io.compression.codec</code>.
   </td>
 </tr>
+<tr>
+  <td><code>spark.io.encryption.enabled</code></td>
+  <td>false</td>
+  <td>
+    Enable IO encryption. It only supports YARN mode.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.io.encryption.keySizeBits</code></td>
+  <td>128</td>
+  <td>
+    IO encryption key size in bits. The valid number includes 128, 192 and 256.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.io.encryption.keygen.algorithm</code></td>
+  <td>HmacSHA1</td>
+  <td>
+    The algorithm to generate the key used by IO encryption. The supported algorithms are
+    described in the KeyGenerator section of the Java Cryptography Architecture Standard Algorithm
+    Name Documentation.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.io.crypto.cipher.transformation</code></td>
+  <td>AES/CTR/NoPadding</td>
+  <td>
+    Cipher transformation for IO encryption. The cipher transformation name is identical to the
+    transformations described in the Cipher section of the Java Cryptography Architecture
+    Standard Algorithm Name Documentation. Currently only "AES/CTR/NoPadding" algorithm is
```

Comment:
If only a single value is supported, maybe just not mention this in the documentation (and mark the config as `.internal()` in the code).